baking/src/tezos_baking/tezos_setup_wizard.py

```diff
@@ -430,16 +435,69 @@ def hashes_comply(s1, s2):
                             and artifact["history_mode"] == "full"
                         )
                     )
-                    and hashes_comply(
-                        artifact["tezos_version"]["commit_info"]["commit_hash"],
-                        node_version_hash,
-                    ),
+                    and pred(artifact),
                     iter(snapshot_array),
                 ),
-                {"url": None, "block_hash": None},
+                None,
+            )
+
+        def hashes_comply(s1, s2):
+            return s1.startswith(s2) or s2.startswith(s1)
+
+        node_version_hash = get_node_version_hash()
+
+        exact_version_pred = (
+            lambda artifact, node_version_hash=node_version_hash: hashes_comply(
+                artifact["tezos_version"]["commit_info"]["commit_hash"],
+                node_version_hash,
             )
+        )
+
+        major_version, _ = get_node_version()
+
+        major_version_pred = (
```

Comment:
This predicate doesn't check 2 things:

1. That `minor_version` is less than the current version. i.e. the wizard can pick the snapshot generated by `17.3` `octez-node` when user only have `17.2` installed, and there is a possibility that they are incompatible

2. it doesn't check that `snapshot_version` is equal to the `compatible_snapshot_version`: it's still a possible case when accepted `snapshot_version` is different between the binaries with the same major version